Code/Tools/AssetProcessor/native/ui/BuilderDataItem.cpp

```diff
@@ -60,17 +65,33 @@ namespace AssetProcessor
         return m_parent;
     }
 
+    AZStd::shared_ptr<BuilderDataItem> BuilderDataItem::InsertChild(AZStd::shared_ptr<BuilderDataItem>&& itemToBeInserted)
+    {
+        //! It's the caller's responsibility to ensure the builder has a unique name
+        if (m_itemType != ItemType::InvisibleRoot ||
+            (itemToBeInserted->GetItemType() != ItemType::InvisibleRoot && itemToBeInserted->GetItemType() != ItemType::Builder) ||
+            itemToBeInserted->GetParent().lock().get() != this || m_childNameToIndex.contains(itemToBeInserted->GetName()))
+        {
+            return nullptr;
+        }
+
+        m_children.push_back(itemToBeInserted);
```

Comment:
I wonder if this operation might not be safe depending on what version of push_back() is used. If the item is moved to the m_children container, then it will be cleared out and line 79 might fail. Maybe I am wrong, just concerned.

Reply:
`itemToBeInserted` is an lvalue, so this will call the `push_back(const T&)` overload. I [noted this as well](https://github.com/o3de/o3de/pull/11005#discussion_r934866434).

Reply:
Thanks, that's a good point. I will use `AZStd::move()` so that I am putting the item into vector using `push_back(T&&)` and reorder the lines. 

However, since the parameter `itemToBeInserted` is of type `AZStd::shared_ptr<BuilderDataItem>&&`, wouldn't the `push_back(T&&)` be called here?

Reply:
solved by reordering the two lines.